### ISSUE TITLE
TPOTBase.score doesn't impute values like TPOTBase.predict does

### ISSUE THREAD

**Issue by villasv** [EpistasisLab/tpot#682]
> TPOT's imputation step isn't stored in the internal pipelines, but it should apply imputation when you call any functions like fit, predict, score, etc.

Here's the code inside `TPOTBase(BaseEstimator).predict`

```
features = features.astype(np.float64)

if np.any(np.isnan(features)):
    self._imputed = True
    features = self._impute_values(features)
else:
    self._imputed = False
return self.fitted_pipeline_.predict(features)
```

So yeah, it will use the `self._fitted_imputer` when predicting because of the `self._impute_values` call.

But now here's the code inside `TPOTBase(BaseEstimator).score`:

```
score = SCORERS[self.scoring_function](
            self.fitted_pipeline_,
            testing_features.astype(np.float64),
            testing_target.astype(np.float64)
)
```

It simply forwards the `testing_features` without using the `self._fitted_imputer`, so unlike I expected, it won't do imputation automatically when using the `score` method.

**Comment by weixuanfu** [comment]
Thank you for reporting this bug. I will push a fix to dev branch soon.

**Comment by weixuanfu** [comment]
I close this issue since I merged the fix to TPOT 0.9.3. Please feel free to reopen this issue if you have any questions.
